ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp

```diff
@@ -7,10 +7,24 @@
 #include "Universal_System/roomsystem.h"
 #include "Universal_System/mathnc.h" // enigma_user::clamp
 
+#include "Universal_System/Extensions/DataStructures/include.h"
+#include "Universal_System/Instances/instance_system.h"
+#include "Universal_System/Instances/instance.h"
+
 #include <chrono> // std::chrono::microseconds
 #include <thread> // sleep_for
 
 namespace enigma {
+  namespace extension_cast {
+    extension_steamworks *as_extension_steamworks(object_basic*);
```

Comment:
@RobertBColton 

This is important for casting to the right tier, but I don't fully understand this as I added the ``myevent_asyncsteam();`` manually!!

Reply:
`object_locals` is the actual class generated by the compiler and it multiply inherits `object_basic` which is "pure" and can't be instantiated, as well as `extension_steamworks` and any other extension interfaces from the `.ey` files.

Because the layout of `object_locals` is not known due to codegen, it's not known where in `object_locals` layout the `extension_steamworks` is.

It basically just casts from the one pure interface `object_basic` to the other interface you `extension_steamworks`:
`extension_steamworks *as_extension_steamworks(object_basic* inst) { return (object_locals*) inst; }`

Reply:
@RobertBColton All these interfaces inherit from object_locals, right?

Reply:
Other way around, `object_locals` inherits the pure interface `object_basic` as well as your extension interface `extension_steamworks`

Reply:
I may need to take a moment to understand this theory, but anyway cooool 🙌